it/it-tests/src/test/java/it/analysis/IssueJsonReportTest.java

```diff
@@ -39,6 +46,69 @@ public void resetData() {
   }
 
   @Test
+  public void issue_line() throws IOException {
+    orchestrator.getServer().restoreProfile(getResource("one-issue-per-line.xml"));
+    orchestrator.getServer().provisionProject("sample", "xoo-sample");
+    orchestrator.getServer().associateProjectToQualityProfile("sample", "xoo", "one-issue-per-line");
+
+    File projectDir = ItUtils.projectDir("shared/xoo-sample");
+    SonarRunner runner = SonarRunner.create(projectDir,
+      "sonar.analysis.mode", "issues",
+      "sonar.verbose", "true",
+      "sonar.report.export.path", "sonar-report.json");
+    BuildResult result = orchestrator.executeBuild(runner);
+    assertThat(ItUtils.countIssuesInJsonReport(result, true)).isEqualTo(17);
+
+    JSONObject obj = ItUtils.getJSONReport(result);
+    JSONArray issues = (JSONArray) obj.get("issues");
+    for (Object issue : issues) {
+      JSONObject jsonIssue = (JSONObject) issue;
+      assertThat(jsonIssue.get("startLine")).isNotNull();
+      assertThat(jsonIssue.get("endLine")).isNotNull();
+
+      assertThat(jsonIssue.get("endLine")).isEqualTo(jsonIssue.get("startLine"));
+
+      assertThat(jsonIssue.get("endOffset")).isNotNull();
+      assertThat(jsonIssue.get("startOffset")).isNotNull();
+    }
+
+    List<Long> lineNumbers = new ArrayList<Long>(16);
+    for (long i = 1L; i < 18; i++) {
+      lineNumbers.add(i);
+    }
+    assertThat(issues).extracting("startLine").containsAll(lineNumbers);
+    assertThat(issues).extracting("endLine").containsAll(lineNumbers);
+  }
+
+  @Test
+  public void precise_issue_location() throws IOException {
+    orchestrator.getServer().restoreProfile(getResource("multiline.xml"));
+    orchestrator.getServer().provisionProject("sample-multiline", "xoo-sample");
+    orchestrator.getServer().associateProjectToQualityProfile("sample-multiline", "xoo", "multiline");
+
+    File projectDir = ItUtils.projectDir("shared/xoo-precise-issues");
+    SonarRunner runner = SonarRunner.create(projectDir,
+      "sonar.analysis.mode", "issues",
+      "sonar.verbose", "true",
+      "sonar.report.export.path", "sonar-report.json");
+    BuildResult result = orchestrator.executeBuild(runner);
+    assertThat(ItUtils.countIssuesInJsonReport(result, true)).isEqualTo(2);
+
+    JSONObject obj = ItUtils.getJSONReport(result);
+    JSONArray issues = (JSONArray) obj.get("issues");
+
+    for (Object i : issues) {
+      JSONObject issue = (JSONObject) i;
```

Comment:
These assertions doesn't look very robust in case there is a mix between first and second issue. Would be better to test if this is first or second issue based for example on "line" value then do other assertions without "isIn".